Script/AtomicEditor/editor/Preferences.ts

```diff
@@ -266,9 +265,63 @@ class Preferences {
         }
 
         // Cache the update
+        return prefs;
+    }
+
+    /**
+     * Sets a user preference value in the user settings file located in the project
+     * @param  {string} settingsGroup name of the group the preference lives under
+     * @param  {string} preferenceName name of the preference to set
+     * @param  {number | boolean | string} value value to set
+     */
+    setUserPreference(settingsGroup: string, preferenceName: string, value: number | boolean | string) {
+
+        const prefsFileLoc = ToolCore.toolSystem.project.userPrefsFullPath;
+        const prefs = this.setGenericPreference(prefsFileLoc, settingsGroup, preferenceName, value);
+
+        // Cache the update
         this._cachedProjectPreferences = prefs;
     }
 
+    /**
+     * Sets an editor preference value in the global user settings file
+     * @param  {string} settingsGroup name of the group the preference lives under
+     * @param  {string} preferenceName name of the preference to set
+     * @param  {number | boolean | string} value value to set
+     */
+    setApplicationPreference(settingsGroup: string, preferenceName: string, value: number | boolean | string) {
+
+        const prefsFileLoc = this.getPreferencesFullPath();
+        const prefs = this.setGenericPreference(prefsFileLoc, settingsGroup, preferenceName, value);
+
+        // Cache the update
+        this._prefs = prefs as PreferencesFormat;
+    }
+
+    /**
+     * Return a preference value or the provided default from the global user settings file located in the project
+     * @param  {string} settingsGroup name of the group these settings should fall under
+     * @param  {string} preferenceName name of the preference to retrieve
+     * @param  {number | boolean | string} defaultValue value to return if pref doesn't exist
+     * @return {number|boolean|string}
+     */
+    getApplicationPreference(settingsGroup: string, preferenceName: string, defaultValue?: number): number;
+    getApplicationPreference(settingsGroup: string, preferenceName: string, defaultValue?: string): string;
+    getApplicationPreference(settingsGroup: string, preferenceName: string, defaultValue?: boolean): boolean;
+    getApplicationPreference(settingsGroup: string, preferenceName: string, defaultValue?: any): any {
```

Comment:
These signatures are actually different based upon the ```defaultValue```.  The final signature is actually never seen from the TS side but it's only seen from the JS side which is how overloads work in TS.  Under the covers that return value really could be anything, but the signatures are there to assist with the TS typechecking so it knows that if you pass a boolean default value then the result should be boolean.  I just checked and saw that if you don't pass a default value, it takes the topmost signature and assumes that the result is a number.